File: src/aside/java/org/taktik/icure/logic/impl/PlaceLogicImpl.java
package org.taktik.icure.logic.impl;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.taktik.icure.dao.PlaceDAO;
import org.taktik.icure.entities.Place;
import org.taktik.icure.exceptions.DeletionException;
import org.taktik.icure.logic.PlaceLogic;
import org.taktik.icure.logic.ICureSessionLogic;

import java.util.List;

@Service
public class PlaceLogicImpl extends GenericLogicImpl<Place, PlaceDAO> implements PlaceLogic {

    private PlaceDAO placeDAO;
    private ICureSessionLogic sessionLogic;

    @Override
    public Place createPlace(Place place) {
        return placeDAO.create(place);
    }

    @Override
    public List<String> deletePlace(List<String> ids) throws DeletionException {
        try {
            deleteEntities(ids);
            return ids;
        } catch (Exception e) {
            throw new DeletionException(e.getMessage(), e);
        }
    }

    @Override
    public Place getPlace(String place) {
        return placeDAO.get(place);
    }

    @Override
    public Place modifyPlace(Place place) {
        return placeDAO.save(place);
    }

    @Autowired
    public void setCalendarItemDAO(PlaceDAO place) {
        this.placeDAO = place;
    }

    @Autowired
    public void setSessionLogic(ICureSessionLogic sessionLogic) {
        this.sessionLogic = sessionLogic;
    }

    @Override
    protected PlaceDAO getGenericDAO() {
        return placeDAO;
    }
}

File: src/main/kotlin/org/taktik/icure/asynclogic/impl/PlaceLogicImpl.kt
/*
 *  iCure Data Stack. Copyright (c) 2020 Taktik SA
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU Affero General Public License as
 *     published by the Free Software Foundation, either version 3 of the
 *     License, or (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful, but
 *     WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *     Affero General Public License for more details.
 *
 *     You should have received a copy of the GNU Affero General Public
 *     License along with this program.  If not, see
 *     <https://www.gnu.org/licenses/>.
 */

package org.taktik.icure.asynclogic.impl

import kotlinx.coroutines.flow.Flow
import org.springframework.stereotype.Service
import org.taktik.couchdb.DocIdentifier
import org.taktik.icure.asyncdao.PlaceDAO
import org.taktik.icure.asynclogic.AsyncSessionLogic
import org.taktik.icure.asynclogic.PlaceLogic
import org.taktik.icure.entities.Place
import org.taktik.icure.exceptions.DeletionException

@Service
class PlaceLogicImpl(private val placeDAO: PlaceDAO,
                     private val sessionLogic: AsyncSessionLogic) : GenericLogicImpl<Place, PlaceDAO>(sessionLogic), PlaceLogic {

    override suspend fun createPlace(place: Place) = fix(place) { place ->
        placeDAO.create(place)
    }

    override fun deletePlace(ids: List<String>): Flow<DocIdentifier> {
        return try {
            deleteEntities(ids)
        } catch (e: Exception) {
            throw DeletionException(e.message, e)
        }
    }

    override suspend fun getPlace(place: String): Place? {
        return placeDAO.get(place)
    }

    override suspend fun modifyPlace(place: Place) = fix(place) { place ->
        placeDAO.save(place)
    }

    override fun getGenericDAO(): PlaceDAO {
        return placeDAO
    }
}
